interface/main/tabs/main.php

```diff
@@ -90,15 +90,112 @@ function isEncounterLocked( encounterId ) {
+                    foreach($input as $paramName => $paramVal) {
+                        // gather url param and query string param vals
+                        if($paramName === "url") {
+                            if(strpos($inputURL, "?")) {
+                                // separate url param from first query string param
+                                $urlParamSplit = explode("?", $paramVal);
+                                $inputURLStr = $urlParamSplit[0];
+                                // separate name and val of first query string param, then add to list
+                                $paramSplit = explode("=", $urlParamSplit[1]);
+                                $inputParamsList[$paramSplit[0]] = $paramSplit[1];
+                            } else {
+                                // gather url param
+                                $inputURLStr = $paramVal;
+                            }
+                        } else {
+                            // add name and val of query string param to list
+                            $inputParamsList[$paramName] = $paramVal;
+                            // rebuild url since $_REQUEST['url'] doesn't pull any other params after the first one
+                            $inputURL .= "&".$paramName."=".$paramVal;
+                        }
+                    }
+
+                    // check if each parsed component of input url is in whitelist
+                    $check = true;
+                    if(array_key_exists($inputURLStr, $allowedURLs)) {
+                        $paramsList = $allowedURLs[$inputURLStr];
+                        // make sure input url include required params
+                        if(!array_key_exists($paramsList["required"], $inputParamsList)) {
+                            $check = false;
+                        } else {
+                            foreach($inputParamsList as $inputParamName => $inputParamVal) {
+                                // validate param names
+                                if(in_array($inputParamName, $paramsList)) {
+                                    // validate param vals
+                                    if($inputParamName === "date") {
+                                        // date format: YYYYMMDD
+                                        $year = substr($inputParamVal, 0, 4);
+                                        $month = substr($inputParamVal, 4, 2);
+                                        $day = substr($inputParamVal, 6, 2);
+                                        if(!checkdate($month, $day, $year)) {
+                                            $check = false;
+                                            break;
+                                        }
+                                    } else if(!preg_match($queryPatterns[$inputParamName], $inputParamVal)) {
+                                        $check = false;
+                                        break;
```

Comment:
A lot of conditioning ... Maybe we should have some time to check out ourselves with sample examples

Reply:
Looks good to me,